### ISSUE TITLE
how can i run infer on CPU?

### ISSUE THREAD

**Issue by JiangWeiHn** [Shiyan-cps/PointKAN-pytorch#4]
我需要把模型运行在没有gpu的电脑上，但是我遇到pointnet2_ops_lib\\pointnet2_ops\\_ext-src\\src\\sampling.cpp":83, please report a bug to PyTorch. CPU not supported。有什么好方法么，我试过：

def fps_cpu(points, npoint):
    """纯 PyTorch CPU 版 FPS
    points: (N, 3) 单点云
    返回: (npoint,) 下标
    """
    N, _ = points.shape
    centroids = torch.zeros(npoint, dtype=torch.long)
    distance = torch.ones(N) * 1e10
    farthest = torch.randint(0, N, (1,)).item()
    for i in range(npoint):
        centroids[i] = farthest
        centroid = points[farthest].view(1, 3)
        dist = torch.sum((points - centroid) ** 2, dim=-1)
        mask = dist < distance
        distance[mask] = dist[mask]
        farthest = torch.argmax(distance).item()
    return centroids

def fps_batch(pos, npoint):
    """适配原始 fps 的批接口
    pos: (B, N, 3)
    返回: (B, npoint) 下标
    """
    B, N, _ = pos.shape
    idx_list = []
    for b in range(B):
        idx = fps_cpu(pos[b], npoint)      # (npoint,)
        idx_list.append(idx)
    return torch.stack(idx_list, dim=0)    # (B, npoint)
 运行时间180s,要死，要死，要死！！
